polus-ftl-label-plugin/src/ftl.pyx

```diff
@@ -0,0 +1,966 @@
+
+    __m256i _mm256_set_epi8(char e31,char e30,char e29,char e28,
+                            char e27,char e26,char e25,char e24,
+                            char e23,char e22,char e21,char e20,
+                            char e19,char e18,char e17,char e16,
+                            char e15,char e14,char e13,char e12,
+                            char e11,char e10,char e9, char e8,
+                            char e7, char e6, char e5, char e4,
+                            char e3, char e2, char e1, char e0) nogil
+    
+    __m256i _mm256_set_epi64x(long long e3,long long e2,
+                              long long e1,long long e0) nogil
+
+    __m256i _mm256_set1_epi8(char a) nogil
+
+    __m256i _mm256_set1_epi16(short a) nogil
+
+    __m256i _mm256_set1_epi32(int a) nogil
+
+    __m256i _mm256_set1_epi64x(long long a) nogil
+
+    __m256i _mm256_srlv_epi64(__m256i a, __m256i b) nogil
+
+    __m256i _mm256_and_si256(__m256i a, __m256i b) nogil
+
+    __m256i _mm256_shuffle_epi8(__m256i a, __m256i b) nogil
+
+    __m256i _mm256_add_epi16(__m256i a, __m256i b) nogil
+
+    __m256i _mm256_add_epi32(__m256i a, __m256i b) nogil
+
+    void _mm256_storeu_si256(__m256i* a,__m256i b) nogil
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.cdivision(True)
+@cython.nonecheck(False)
+cdef np.ndarray run_length_encode_16(unsigned char [:] image,tuple shape):
+    """run_length_encode Create run length encoded pixels
+
+    This function will run on a matrix with arbitrary number of dimensions.
+    It is best to use the appropriate run length encoding algorithm for a
+    specified number of dimensions if it exists.
+
+    This function finds consecutive pixels along the final dimension
+    that have non-zero values. Then, the first and last dimension index
+    along with the other coordinates are stored in a numpy array. This
+    results in a 2d numpy array with ndim+1 columns for each column object:
+    [...,last_dim_index_start,last_dim_index_end]
+
+    The input image must be a linear array (i.e. use reshape(-1)).
+    The input image must be np.uint8 or bool data type.
+
+    Note: This function is the main bottleneck in the labeling algorithm.
+        This method generally takes up 50-90% or more of the time required
+        image labeling depending on the size and number of objects in the
+        the image. This function can be made to run faster by specializing
+        for a specific number of dimensions, particularly in the section
+        where values are stored. It may be a good idea to do this for common
+        matrix dimensions (2d and 3d), since the performance improvement
+        would be noticeable.
+
+    Args:
+        image (numpy.ndarray): An n-dimensional image reshaped to a linear
+            array of pixels.
+        shape (tuple): The shape of the image
+        
+    Outputs:
+        numpy.ndarray: Indices of pixel objects
+    """
+
+    ''' Set up AVX vectors '''
```

Comment:
I know that I was being added to review the intrinsic function in ftl.pyx. I should say that the SIMD that I practice in my parallelization is at the level of C++ native language. The compiler then converts my code to machine language object using intrinsic functions used here. I do not have prior experience with these intrinsic functions and can not provide you with the detailed evaluation of this code. However, I will provide you with some comments reflecting my overall thoughts.